experimental/httpclient/http_client.go

```diff
@@ -0,0 +1,21 @@
+package httpclient
+
+import (
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/errorsource"
+)
+
+// New wraps the existing http client constructor and adds the error source middleware
+func New(opts *httpclient.Options) (*http.Client, error) {
+	id := uuid.New()
+	opts.Middlewares = append(opts.Middlewares, errorsource.Middleware(id.String()))
```

Comment:
This will remove all the default middlewares unless they are provided by the caller:

https://github.com/grafana/grafana-plugin-sdk-go/blob/8140bf11e904522b6c91020d2efb12a6b62d6957/backend/httpclient/http_client.go#L15-L17

So a developer should call this function like this:

```go

// ...

exphttpclient "github.com/grafana/grafana-plugin-sdk-go/experimental/httpclient"

// ...

cl, err := exphttpclient.New(&httpclient.Options{
	Middlewares: httpclient.DefaultMiddlewares(),
})
```

:point_up:  This keeps the default middlewares and appends the `errorsource.Middleware` at the end.

However, when using the normal `httpclient` package, a plugin developer usually creates a new http client like this instead:

```go
cl, err := httpclient.New()
```

and expects the default middlewares to be present.

---

I guess what you want to do in this method instead is:

```suggestion
	if len(opts.Middlewares) == 0 {
		opts.Middlewares = httpclient.DefaultMiddlewares()
	}
	opts.Middlewares = append(opts.Middlewares, errorsource.Middleware(id.String()))
```

This will use the `DefaultMiddlewares` if none are provided, otherwise use the ones provided by the user, and append the `errorsource.Middleware` afterwards